### ISSUE TITLE
PaLM: Scaling Language Modeling with Pathways

### ISSUE THREAD

**Issue by nariaki3551** [nariaki3551/library#63]
- year: 2022
- journal: arxiv
- url: https://arxiv.org/pdf/2204.02311
- google scholar: https://scholar.google.co.jp/scholar?hl=ja&as_sdt=0%2C5&q=PaLM%3A+Scaling+Language+Modeling+with+Pathways&btnG=
- scispace: https://typeset.io/library/untitled-collection-6c0g6k75/palm-scaling-language-modeling-with-pathways-pdf-2i5weqpx
- cited: 1107 (3-July-2023)

### Typeset要約

この論文では、Pathways Language Model（PaLM）と呼ばれる新しい言語モデルを紹介します。これは、5,400億個のパラメーターで構成され、高密度に活性化されたトランスフォーマー言語モデルです。このモデルは、複数のTPUポッドにわたって非常に効率的なトレーニングを可能にする新しいMLシステムであるPathwaysを使用して、6144個のTPU v4チップでトレーニングされました。このホワイトペーパーでは、何百もの言語理解と生成のベンチマークで最先端の少ない学習結果を得ることにより、スケーリングのメリットを実証しています。

**Comment by nariaki3551** [comment]
## Section 3

The PaLM pretraining dataset consists of a high-quality corpus of 780 billion tokens that represent a wide range of natural language use cases. The dataset is a mixture of filtered webpages, books, Wikipedia, news articles, source code, and social media conversations. This dataset is based on the datasets used to train LaMDA (Thoppilan et al., 2022) and GLaM (Du et al., 2021). We train all three models on exactly one epoch of the data (shuffled identically for all models) and choose the mixing proportions to avoid repeating data in any subcomponent.

> PaLMの事前学習データセットは、幅広い自然言語の使用例を表す7800億トークンの高品質コーパスから構成されている。このデータセットは、フィルタリングされたウェブページ、書籍、ウィキペディア、ニュース記事、ソースコード、ソーシャルメディアでの会話が混在している。このデータセットは、LaMDA (Thoppilan et al., 2022)とGLaM (Du et al., 2021)の学習に使用されたデータセットをベースにしている。我々は3つのモデル全てを正確に1エポックのデータで訓練し（全てのモデルで同じようにシャッフルされる）、どのサブコンポーネントでもデータの繰り返しを避けるように混合比率を選択する。

In addition to natural language data, the pretraining dataset also contains code. The source code in the pretraining dataset is obtained from open source repositories on GitHub. We filtered the files by the license included in the repository; copyleft licenses were excluded. We filter the files by filename extension to restrict to one of 24 common programming languages, including Java, HTML, Javascript, Python, PHP, C#, XML, C++, and C, which results in 196GB of source code. Further, we remove duplicates based on Levenshtein distance between the files because duplicate files are known to be common in source code repositories (Lopes et al., 2017; Allamanis, 2019).

> 自然言語データに加えて、事前学習データセットにはコードも含まれる。プレトレーニングデータセットのソースコードは、GitHub上のオープンソースリポジトリから入手した。リポジトリに含まれるライセンスによってファイルをフィルタリングし、コピーレフトのライセンスは除外した。ファイル名の拡張子によってファイルをフィルタリングし、Java、HTML、Javascript、Python、PHP、C#、XML、C++、および C を含む 24 の一般的なプログラミング言語のうちの 1 つに制限します。さらに、重複ファイルはソースコードリポジトリでよく見られることが知られているため（Lopes et al., 2017; Allamanis, 2019）、ファイル間のレーベンシュタイン距離に基づいて重複を削除している。

Table 2 lists the proportion of various data sources used to create the final PaLM dataset mixture. We check for data contamination in Section 8 and present an overlap analysis between our training dataset and the evaluation data. Appendix D contains a datasheet (Gebru et al., 2021) with additional information, including a breakdown of language proportions.

> 表2は、最終的な混合PaLMデータセットを作成するために使用された様々なデータソースの割合を示している。セクション8でデータの混入をチェックし、学習データセットと評価データの重複分析を行う。付録Dには、言語比率の内訳を含む追加情報が記載されたデータシート（Gebru et al.

![スクリーンショット 2023-07-03 15 15 44](https://github.com/nariaki3551/library/assets/21208556/59b89894-cd6b-4a04-896e-7f6fa36cbc5b)
